nav2_msgs/srv/LoadMap.srv

```diff
@@ -0,0 +1,22 @@
+# Type definitions
+uint8 TYPE_FILE=0
+uint8 TYPE_URL=1
+
+# Type of map resource
+uint8 type
+
+# ID of map resource
+string map_id
+
+---
+
+# Result code defintions
+uint8 RESULT_SUCCESS=0
+uint8 RESULT_MAP_DOES_NOT_EXIST=1
+uint8 RESULT_INVALID_MAP_DATA=2
+uint8 RESULT_INVALID_MAP_METADATA=2
+uint8 RESULT_UNDEFINED_FAILURE=255
+
+# Returned map is only valid if result equals RESULT_SUCCESS
+nav_msgs/OccupancyGrid map
```

Comment:
Huh, while the use-case I intended to cover was the LoadMap with the intention of changing the map server's map, I suppose its also useful to someone to get a map as well.

Maybe separately, maybe in this same service have a bool to change the map. Right now its loading the map and returning it, but its not changing the map being broadcasted by the server since the publish is only happening once at activation. At minimum at the end of the callback it should `occ_pub_->publish(*msg_);` so its being broadcasted by default.

Reply:
Good point, I can add the publish. I based this on the LoadMap service in the [upstream PR](https://github.com/ros-planning/navigation/pull/461) you pointed at in the original issue #239 

That one may never merge but I used the service as a starting point

Reply:
That PR publishes in the loading step https://github.com/ros-planning/navigation/blob/6da759231c37f22fb932e51beab0025d68497fee/map_server/src/main.cpp#L159 which differs from ours

Reply:
Done